python/xrprimer/utils/visualization_utils.py

```diff
@@ -72,3 +89,108 @@ def fix_arr_shape(
                          f' getting {array.shape}.')
             raise ValueError
     return array
+
+
+def check_data_len(mframe_point_data: Union[np.ndarray, None],
+                   mframe_line_data: Union[np.ndarray, None],
+                   background_len: int, logger: logging.Logger) -> int:
+    """Check length of data. Return the length if all data match.
+
+    Args:
+        mframe_point_data (Union[np.ndarray, None]):
+            Mulit-frame point data.
+        mframe_line_data (Union[np.ndarray, None]):
+            Mulit-frame line data.
+        background_len (int):
+            Length of background.
+        logger (logging.Logger):
+            Logger for logging.
+
+    Raises:
+        ValueError:
+            Length of point_data,
+            line_data and background do not match.
+
+    Returns:
+        int: Length of data.
+    """
+    empty_data = True
+    len_list = []
+    if background_len > 0:
+        len_list.append(background_len)
+    if mframe_point_data is not None:
+        empty_data = False
+        len_list.append(mframe_point_data.shape[0])
+    if mframe_line_data is not None:
+        empty_data = False
+        len_list.append(mframe_line_data.shape[0])
+    if empty_data:
+        logger.error('Please pass point_data or line_data or both.')
+        raise ValueError
+    ref_len = len_list[0]
+    len_correct = True
+    for tmp_len in len_list:
+        if tmp_len != ref_len:
+            len_correct = False
+            break
+    if not len_correct:
+        logger.error('Length of point_data, line_data' +
+                     ' and background do not match.')
+        raise ValueError
+    else:
+        return ref_len
+
+
+def check_output_path(output_path: str, overwrite: bool,
+                      logger: logging.Logger) -> None:
+    """Check output path.
+
+    Args:
+        output_path (str):
+            Path to output.
+        overwrite (bool):
+            Whether to overwrite existing files.
+        logger (logging.Logger):
+            Logger for logging.
+
+    Raises:
+        FileNotFoundError: Parent of output_path doesn't exist.
+        FileExistsError: output_path exists and overwrite not enabled.
+    """
+    existence = check_path_existence(output_path)
+    if existence == Existence.MissingParent:
+        logger.error(f'Parent of {output_path} doesn\'t exist.')
+        raise FileNotFoundError
+    elif (existence == Existence.DirectoryExistNotEmpty
+          or existence == Existence.FileExist) and not overwrite:
+        logger.error(f'{output_path} exists and overwrite not enabled.')
+        raise FileExistsError
+    if not check_path_suffix(output_path, '.mp4'):
+        os.makedirs(output_path, exist_ok=True)
+
+
+def rotate_3d_data(data: np.ndarray, left_mm_rotmat: np.ndarray,
```

Comment:
What about renaming it to `rotate_3d_points` and placing it under `xrprimer/transform/point`?

Reply:
Fixed in [f973972](https://github.com/openxrlab/xrprimer/pull/39/commits/f9739722b97083b69836a4dc395c9cbabe4e62a6).